src/main/java/gui/viewmodel/ViewModelFactory.java

```diff
@@ -4,37 +4,58 @@
 import services.servicesfactory.ObservableRepositoryFactory;
 import datagateway.event.ObservableEventRepository;
 import datagateway.task.ObservableTaskRepository;
+import services.servicesfactory.ServicesFactory;
 
 public class ViewModelFactory {
 
-    private final MonthlyCalendarViewModel monthlyCalendarViewModel;
-    private final WeeklyCalendarViewModel weeklyCalendarViewModel;
-    private final TodoListPageViewModel todoListPageViewModel;
-    private final AddTaskPageViewModel addTaskPageViewModel;
-
-    public ViewModelFactory(ObservableEventRepository eventRepository,
-                            ObservableTaskRepository taskRepository) {
-        this.monthlyCalendarViewModel = new MonthlyCalendarViewModel(eventRepository);
-        this.weeklyCalendarViewModel = new WeeklyCalendarViewModel(eventRepository);
-        this.todoListPageViewModel = new TodoListPageViewModel(taskRepository);
-        this.addTaskPageViewModel = new AddTaskPageViewModel(taskRepository);
-    }
+    private final ServicesFactory servicesFactory;
+    private final ObservableEventRepository eventRepository;
+    private final ObservableTaskRepository taskRepository;
+
+    private MonthlyCalendarViewModel monthlyCalendarViewModel;
+    private WeeklyCalendarViewModel weeklyCalendarViewModel;
+    private TodoListPageViewModel todoListPageViewModel;
+    private AddTaskPageViewModel addTaskPageViewModel;
 
-    public ViewModelFactory(ObservableRepositoryFactory repositoryFactory) {
-        this(repositoryFactory.makeEventRepository(), repositoryFactory.makeTaskRepository());
+    public ViewModelFactory(ObservableRepositoryFactory repositoryFactory, ServicesFactory servicesFactory) {
+        this.servicesFactory = servicesFactory;
+        eventRepository = repositoryFactory.makeEventRepository();
+        taskRepository = repositoryFactory.makeTaskRepository();
     }
 
     public MonthlyCalendarViewModel getMonthlyCalendarViewModel() {
+        if (monthlyCalendarViewModel == null) {
```

Comment:
Why having this condition check?

Reply:
becoz if it is null, it means that it has not been initialized. if it is not null, we will return the one that has been initialized. this ensures that we are always getting the same view model